### PR TITLE
Adds rational-quadratic kernel and minor clean-up of notation

### PR DESCRIPTION
**Pull request by st--**
- renames square_dist and euclid_dist to scaled_square_dist and scaled_euclid_dist to be more consistent about what they actually return, and adds docstrings
- adds SquaredExponential as alias to RBF
- adds RationalQuadratic kernel
- includes a test for Rational Quadratic (\alpha->\infty) -> Squared Exponential

### CODE REVIEW

File: gpflow/kernels.py [hunk 1]
```diff
@@ -232,7 +232,11 @@ def __init__(self, input_dim, variance=1.0, lengthscales=None,
             self.ARD = False
 
     @params_as_tensors
-    def square_dist(self, X, X2):
+    def scaled_square_dist(self, X, X2):
+        """
+        Returns ((X - X2ᵀ)/lengthscales)².
+        Due to the implementation, the result may actually be negative.
+        """
```
Comment (markvdw) [inline gpflow/kernels.py]: I'm against renaming this method. Second opinion of @jameshensman? If James agrees to the renaming, I would like to insist on deprecating `square_dist()` with a warning for the time being.

Also, with "the result may be actually be negative", you mean due to floating point precision, right? Could you clarify this in the docstring. Right now it just seems like the implementation is wrong.

File: gpflow/kernels.py [hunk 2]
```diff
@@ -248,8 +252,11 @@ def square_dist(self, X, X2):
         return dist
 
 
-    def euclid_dist(self, X, X2):
-        r2 = self.square_dist(X, X2)
+    def scaled_euclid_dist(self, X, X2):
+        """
+        Returns |(X - X2ᵀ)/lengthscales| (L2-norm).
+        """
+        r2 = self.scaled_square_dist(X, X2)
```
Comment (markvdw) [inline gpflow/kernels.py]: Idem above.

File: gpflow/kernels.py [hunk 3]
```diff
@@ -266,7 +273,35 @@ class RBF(Stationary):
...
+    """
+    Rational Quadratic kernel,
+
+    k(r) = σ² (1 + r² / 2αℓ²)^(-α)
+
+    σ² : variance
+    ℓ  : lengthscales
+    α  : alpha, determines relative weighting of small-scale and large-scale fluctuations
+
+    For α→ ∞, the RQ kernel becomes equivalent to the squared exponential.
```
Comment (markvdw) [inline gpflow/kernels.py]: Add space after alpha. Very nitpicky, but the look of the lack of space makes me sqirm :)

Also, I would do it myself, but then my review would be attached to an old commit.

Comment (st--) [reply gpflow/kernels.py]: Funnily enough, there is no space *after* the \alpha either... it's just a quirk of the font :roll_eyes: but I'm happy to add a space beforehand if that makes you happier

File: gpflow/kernels.py (hunk 1 above)
Comment (st--) [reply gpflow/kernels.py]: Happy to add deprecated methods.

**Pull request merged**